app/auth-portal/src/content/tutorial/08-asset_authoring.md

```diff
@@ -0,0 +1,442 @@
+code and replace it with the following:
+
+```js
+async function generateCode(input: Input): Promise<Output> {
+  const object = {
+    AvailabilityZone: input.domain?.availabilityZone,
+    Iops: input.domain?.iops,
+    VolumeType: input.domain?.volumeType,
+    Size: input.domain?.size,
+    Encrypted: false, // We are defaulted to non-encryped right now
+  };
+
+  return {
+    format: "json",
+    code: JSON.stringify(object, null, "\t"),
+  };
+}
+```
+
+Use the `Execute` button to be able to save and update the function attachment.
+
+![codegen-func-attachment.png](/tutorial-img/08-asset-authoring/codegen-func-attachment.png)
+
+*~> Notice, a display name is necessary for the function, so I have made the display name the same as the function name*
+
+We are now able to use this code generation function to be able to qualify and create the resource.
+
+### Authoring a Qualification Function
+
+A qualification function is used to understand if the resource we are attempting to create is going to work as expected.
+We can do this using something like a `dry-run` functionality of a CLI. Select the asset we just created
+and use the `Attach Function` button and select `New function`.
+
+I will choose `Qualification` and set the name to `qualifyAwsEbsVolume` and then create it. When we click the button,
+we get a pre-generated code snippet in the function editor. Remove the pre-generated
+code and replace it with the following:
+
+```js
+async function qualification(input: Input): Promise<Output> {
+  // This is the name of the Code Generation function we created above!  
+  const code = input.code?.["awsEc2EbsVolumeJSON"]?.code;
+  if (!code) {
+    return {
+      result: "failure",
+      message: "component doesn't have JSON representation",
+    };
+  }
+
+  if (!input.domain?.region) {
+    return {
+      result: "failure",
+      message: "component doesn't have a region set",
+    };
+  }
+  
+  const child = await siExec.waitUntilEnd("aws", [
+    "ec2",
+    "create-volume",
+    "--region",
+    input.domain.region,
+    "--dry-run",
+    "--cli-input-json",
+    code,
+  ]);
+
+  const success = child.stderr.includes("An error occurred (DryRunOperation)");
+
+  return {
+    result: success ? "success" : "failure",
+    message: success ? "component qualified" : child.stderr,
+  };
+}
+```
+
+We can use the codegen function that we wrote earlier, and we can shell out to the AWS CLI to be able to qualify that the
+component is correct before we try and create it. We can execute that function to update the function mapping and then
+the qualification is ready for us to use.
+
+### Authoring Action Functions
+
+Action functions are a way of being able to attach behavior to our components. The types of actions we can do are:
+
+* Create
+* Delete
+* Refresh
+
+Let's start with authoring create, delete and refresh functions. It's important to set the `entrypoint` of the function attributes
+to match the name of the action.
+
```

Comment:
reproduce the same pattern above, where you tell the user to create a new function, choose action, delete the snippet